orchestration/dagster_orchestration/hca_orchestration/resources/beam.py

```diff
@@ -1,65 +1,87 @@
+class DataflowCloudConfig:
     project: str
-    temp_location: str
-    subnet_name: str
     service_account: str
+    subnet_name: str
+    region: str
+    worker_machine_type: str
+    starting_workers: int
+    max_workers: int
+
+    # fields computed from provided params
+    subnetwork: str = field(init=False)
+
+    def __post_init__(self):
+        self.subnetwork = '/'.join([
+            'regions',
+            self.cloud_config.region,
+            'subnetworks',
+            self.cloud_config.subnet_name,
+        ])
+
+
+@dataclass
+class DataflowBeamRunner:
+    cloud_config: DataflowCloudConfig
+    kubernetes_service_account: str
+    temp_bucket: str
     image_name: str
     image_version: str
     namespace: str
+    logger: DagsterLogManager
 
     def run(
         self,
         job_name: str,
         input_prefix: str,
-        output_prefix: str,
-        context: AbstractComputeExecutionContext
+        output_prefix: str
     ) -> None:
+        args_dict = {
+            'runner': 'dataflow',
+            'inputPrefix': input_prefix,
+            'outputPrefix': output_prefix,
+            'project': self.cloud_config.project,
+            'region': self.cloud_config.region,
+            'tempLocation': f'gs://{self.temp_bucket}/dataflow',
+            'subnetwork': self.cloud_config.subnetwork,
+            'serviceAccount': self.cloud_config.service_account,
+            'workerMachineType': self.cloud_config.worker_machine_type,
+            'autoscalingAlgorithm': 'THROUGHPUT_BASED',
+            'numWorkers': str(self.cloud_config.starting_workers),
+            'maxNumWorkers': str(self.cloud_config.max_workers),
+            'experiments': 'shuffle_mode=service',
+        }
+
+        # process the args dict into dataflow flags
         args = [
-            '--runner=dataflow',
-            f"--inputPrefix={input_prefix}",
-            f"--outputPrefix={output_prefix}",
-            f"--project={self.project}",
-            "--region=us-central1",
-            f"--tempLocation={self.temp_location}",
-            f"--subnetwork=regions/us-central1/subnetworks/{self.subnet_name}",
-            f"--serviceAccount={self.service_account}",
-            "--workerMachineType=n1-standard-4",
-            "--autoscalingAlgorithm=THROUGHPUT_BASED",
-            "--numWorkers=4",
-            "--maxNumWorkers=16",
-            "--experiments=shuffle_mode=service"
+            f'--{field_name}={value}'
+            for field_name, value
+            in args_dict.items()
         ]
 
         image_name = f"{self.image_name}:{self.image_version}"  # {context.solid_config['version']}"
-        job = self.dispatch_k8s_job(self.namespace, image_name, job_name, args, context)
-        context.log.info("Dataflow job started")
+        job = self.dispatch_k8s_job(image_name, job_name, args)
+        self.logger.info("Dataflow job started")
 
-        DataflowBeamRunner.get_job_status(job.metadata.name, self.namespace)
```

Comment:
this line appeared to just fetch the status and toss it into the void, let me know if this is necessary for waiting on a particular external state or something similar and i'll put it back with an explanatory comment

Reply:
I think we added this to aid in debugging as were developing this, should be good to remove.